subsys/bluetooth/mesh/shell.c

```diff
@@ -1932,6 +2350,128 @@ static int cmd_provision_adv(const struct shell *shell, size_t argc,
 
 	return 0;
 }
+
+static int cmd_prov_close_link(const struct shell *shell, size_t argc,
+			     char *argv[])
+{
+	pb_adv.link_close(PROV_BEARER_LINK_STATUS_FAIL);
+
+	return 0;
+}
+
+static int cmd_prov_static_oob(const struct shell *shell, size_t argc,
+			     char *argv[])
+{
+	size_t len;
+	int err;
+
+	if (argc < 2) {
+		return -EINVAL;
+	} else {
+		uint8_t static_key[16];
+
+		len = hex2bin(argv[1], strlen(argv[1]),
+					      static_key, 16);
+
+		err = bt_mesh_auth_method_set_static(static_key, len);
+	}
+
+	if (err) {
+		shell_error(shell, "Static OOB value set Failed (err %d)", err);
+	} else {
+		shell_print(shell, "Static OOB value set (size %u)", len);
+	}
+
+	return 0;
+}
+
+static int cmd_prov_input_oob(const struct shell *shell, size_t argc,
+			     char *argv[])
+{
+	int err;
+	size_t size;
+
+	if (argc < 3) {
+		return -EINVAL;
+	} else {
+		bt_mesh_input_action_t act;
+
+		act = (bt_mesh_input_action_t)str2u8(argv[1]);
+		size = str2u8(argv[2]);
+
+		err = bt_mesh_auth_method_set_input(act, size);
+	}
+
+	if (err) {
+		shell_error(shell, "Input OOB Set Failed (err %d)", err);
+	} else {
+		shell_print(shell, "Input OOB value set (size %u)", size);
+	}
+
+	return 0;
+}
+
+static int cmd_prov_output_oob(const struct shell *shell, size_t argc,
+			     char *argv[])
+{
+	int err;
+	size_t size;
+
+	if (argc < 3) {
+		return -EINVAL;
+	} else {
+		bt_mesh_output_action_t act;
+
+		act = (bt_mesh_output_action_t)str2u8(argv[1]);
+		size = str2u8(argv[2]);
+
+		err = bt_mesh_auth_method_set_output(act, size);
+	}
+
+	if (err) {
+		shell_error(shell, "Output OOB Set Failed (err %d)", err);
+	} else {
+		shell_print(shell, "Output OOB value set (size %u)", size);
+	}
+
+	return 0;
+}
+
+static int cmd_prov_no_oob(const struct shell *shell, size_t argc,
+			     char *argv[])
+{
+	bt_mesh_auth_method_set_none();
+
+	shell_print(shell, "None OOB value set");
+
+	return 0;
+}
+
+static int cmd_prov_public_key(const struct shell *shell, size_t argc,
+			     char *argv[])
+{
+	int err;
+	uint8_t pub_key[64];
+
+	/* Replace Own Public Key Here */
+	static const uint8_t *pub_str = {
+		"F465E43FF23D3F1B9DC7DFC04DA87581"
+		"84DBC966204796ECCF0D6CF5E16500CC"
+		"0201D048BCBBD899EEEFC424164E33C2"
+		"01C2B010CA6B4D43A8A155CAD8ECB279"
+	};
```

Comment:
I don't know how useful this will be. The public key should be different for every device, but this locks our shell application image to a single key. This command should at least accept a parameter to set this key